user/views.py

```diff
@@ -0,0 +1,30 @@
+from rest_framework import status, viewsets
+from rest_framework.exceptions import AuthenticationFailed
+from rest_framework.permissions import AllowAny
+from rest_framework.response import Response
+
+from user.models import User
+from user.serializers import UserSerializer
+from user.utils import generate_access_token
+
+
+class UserViewSet(viewsets.ViewSet):
+    permission_classes = (AllowAny,)
+
+    def sign_up(self, request):
+        user_serializer = UserSerializer(data=request.data)
+        user_serializer.is_valid(raise_exception=True)
+
+        user_serializer.save()
+        return Response(user_serializer.data, status.HTTP_201_CREATED)
+
+    def sign_in(self, request):
+        user = User.objects.filter(username=request.data['username'])
+
+        if not user.exists:
```

Comment:
**Suggestion:** In the 'sign_in' method, the 'exists' method is being called on a QuerySet object but it's not being invoked as a method. This will always evaluate to True, even if the user doesn't exist. This should be corrected to 'exists()'.
```suggestion
        user = User.objects.filter(username=request.data['username'])
        if not user.exists():
            raise AuthenticationFailed('user not found')
```